File: src/TypeFormatter/BooleanTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { BooleanType } from "../Type/BooleanType.js";

export class BooleanTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof BooleanType;
    }
    public getDefinition(type: BooleanType): Definition {
        return { type: "boolean" };
    }
    public getChildren(type: BooleanType): BaseType[] {
        return [];
    }
}

File: internal/formatter/boolean_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// BooleanTypeFormatter mirrors src/TypeFormatter/BooleanTypeFormatter.ts.
type BooleanTypeFormatter struct{}

func NewBooleanTypeFormatter() *BooleanTypeFormatter { return &BooleanTypeFormatter{} }

func (f *BooleanTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.BooleanType)
	return ok
}

func (f *BooleanTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{Type: "boolean"}
}

func (f *BooleanTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
